core-test/src/test/scala/org/bitcoins/core/protocol/tlv/EventDescriptorTest.scala

```diff
@@ -0,0 +1,39 @@
+package org.bitcoins.core.protocol.tlv
+
+import org.bitcoins.core.number.{Int32, UInt16, UInt32}
+import org.bitcoins.testkit.util.BitcoinSUnitTest
+
+class EventDescriptorTest extends BitcoinSUnitTest {
+
+  behavior of "EventDescriptor"
+
+  it must "create an enumerated event" in {
+    val outcomes = Vector("Democrat_win", "Republican_win", "other")
+    val enumEventDescriptorV0TLV = EnumEventDescriptorV0TLV(outcomes)
+
+    assert(enumEventDescriptorV0TLV.noncesNeeded == 1)
+  }
+
+  it must "create a range event" in {
+    val rangeEventDescriptorV0TLV =
+      RangeEventDescriptorV0TLV(start = Int32(-2),
+                                count = UInt32(4),
+                                step = UInt16.one,
+                                unit = "",
+                                precision = Int32.one)
+
+    assert(rangeEventDescriptorV0TLV.outcomes == Vector("-2", "-1", "0", "1"))
+  }
+
+  it must "create a signed digit decomposition event" in {
+    val descriptor =
+      UnsignedDigitDecompositionEventDescriptor(base = UInt16(10),
```

Comment:
this says signed but you made an unsigned.


Should also add a tests for both signed and unsigned as well as multiple different bases (2, 10, 16) and number of digits.